services/app-api/handlers/forms/get.ts

```diff
@@ -0,0 +1,60 @@
+import handler from "../handler-lib";
+import dynamoDb from "../../utils/dynamo/dynamodb-lib";
+import { StatusCodes } from "../../utils/types/types";
+import { NO_KEY_ERROR_MESSAGE } from "../../utils/constants/constants";
+import { Key } from "aws-sdk/clients/dynamodb";
+
+export const getForm = handler(async (event, _context) => {
+  if (!event?.pathParameters?.formName! || !event?.pathParameters?.formId!) {
+    throw new Error(NO_KEY_ERROR_MESSAGE);
+  }
+  const queryParams = {
+    TableName: process.env.FORM_TABLE_NAME!,
+    KeyConditionExpression: "#formName = :formName AND #formId = :formId",
+    ExpressionAttributeValues: {
+      ":formName": event.pathParameters.formName,
+      ":formId": event.pathParameters.formId,
+    },
+    ExpressionAttributeNames: {
+      "#formName": "formName",
+      "#formId": "formId",
+    },
+  };
+  const queryResponse = await dynamoDb.query(queryParams);
+  const responseBody = queryResponse.Items![0] ?? {};
+  return {
+    status: StatusCodes.SUCCESS,
+    body: responseBody,
+  };
+});
+
+export const getLatestForm = handler(async (_event, _context) => {
+  const queryParams = {
+    TableName: process.env.FORM_TABLE_NAME!,
+    ExclusiveStartKey: undefined as Key | undefined,
+  };
+
+  const scannedResults: any[] = [];
+  let queryValue;
+
+  do {
+    queryValue = await dynamoDb.scan(queryParams);
+    if (queryValue?.Items) {
+      scannedResults.push(...queryValue.Items);
+    }
+    queryParams.ExclusiveStartKey = queryValue.LastEvaluatedKey;
+  } while (queryValue.LastEvaluatedKey !== undefined);
+
+  const latestFormTimestamp = Math.max(
+    ...scannedResults.map((item) => parseInt(item.createdAt))
+  );
+
+  const latestForm = scannedResults.filter(
+    (item) => parseInt(item.createdAt) === latestFormTimestamp
+  );
+
+  return {
+    status: StatusCodes.SUCCESS,
+    body: latestForm[0],
+  };
+});
```

Comment:
i might be thinking about this wrong, but i don't think we'll ever need to get the latest form. the latest form will always be the form defined in the code, and when we create a new report on the dashboard, we can just use that 'local' form as the form that gets associated to the new report. then when we pull down reportData (on entry into form) we also pull down the associated form and populate the entire report fill-out and validation experience from that. so we wouldn't ever need the latest form -- just the form as specified by name/id so we could pull it down on a report-by-report basis.

Reply:
removed